### ISSUE TITLE
[CA1508] False positive for "string.IsNullOrWhiteSpace(stringValue)"

### ISSUE THREAD

**Issue by TobiasSekan** [dotnet/roslyn#44541]
**Version Used**: 
Microsoft Visual Studio Community 2019 Preview
Version 16.7.0 Preview 1.0

**Steps to Reproduce**:
```csharp
public void MakeCoolThings(object value)
{
    if(value is string stringValue)
    {
        // CA1508 -> string.IsNullOrWhiteSpace(stringValue) is always false
        if(string.IsNullOrWhiteSpace(stringValue))
        {
            // do cool stuff here
        }
    }

    // other cool stuff
}

// test with
// MakeCoolThings("\n\n");
```

**Expected Behavior**:
no CA1508

**Actual Behavior**:
CA1508 -> string.IsNullOrWhiteSpace(stringValue) is always false

**Comment by Youssef1313** [comment]
Is this related to dotnet/roslyn-analyzers ?

**Comment by Youssef1313** [comment]
I don't get the warning on my machine.
Tested with Microsoft.CodeAnalysis.FxCopAnalyzers v2.9.6
![image](https://user-images.githubusercontent.com/31348972/82839915-37382000-9ed1-11ea-8053-b0380ccfd59c.png)

**Comment by jinujoseph** [comment]
cc @mavasani
